src/entities/QuarterBudget/model.ts

```diff
@@ -172,27 +186,43 @@ export default class QuarterBudgetModel extends Model<QuarterBudgetAttributes> {
     })
   }
 
-  static parseBudget(result: BudgetQueryResult[]): Budget {
-    const categoriesResults: Record<string, CategoryBudget> = {}
+  static parseBudget(result: BudgetQueryResult[]): Budget[] {
```

Comment:
I'm finding this harder to follow than the previous `parseBudget` function.

How about having a `parseBudgets` function that groups the category budgets by id, and then calls the previous `parseBudget` on each group?


```typescript
static parseBudgets(results: BudgetQueryResult[]): Budget[] {
    const budgetsByQuarter = this.groupBudgetResultsByQuarter(results)
    const budgets: Budget[] = []

    budgetsByQuarter.forEach((resultsForQuarter) => {
      const budget = this.parseBudget(resultsForQuarter)
      budgets.push(budget)
    })

    return budgets
  }

  static groupBudgetResultsByQuarter(budgetResults: BudgetQueryResult[]): Map<string, BudgetQueryResult[]> {
    const groupedResults = new Map<string, BudgetQueryResult[]>()

    budgetResults.forEach((result) => {
      const { id } = result
      const group = groupedResults.get(id)

      if (group) {
        group.push(result)
      } else {
        groupedResults.set(id, [result])
      }
    })

    return groupedResults
  }
  ```
  
  Then you just use `return this.parseBudgets(results)`
  
  And you could leave the parseBudget in the other functions

Reply:
Agree parseBudget should only return one budget.

Reply:
Not sure about having two more functions only used on getAllBudgets though. I'll try to gather all that info in one function and see if it is understandable.